<launch>
    <!-- 定义命令行参数 -->
    <arg name="swarm_num_uav" default="3"/>
    <!-- 无人机初始位置和终点位置 -->
	<arg name="uav1_init_x" default="0.0"/>
    <arg name="uav1_init_y" default="0.0"/>
    <arg name="uav1_init_z" default="3.0"/>
    <!-- <arg name="uav1_init_z" default="20.0"/> -->

	<arg name="uav1_end_x" default="10.0"/>
    <arg name="uav1_end_y" default="-20.0"/>
    <arg name="uav1_end_z" default="9.0"/>

    <arg name="uav1_offset_x" default="0.0"/>
    <arg name="uav1_offset_y" default="0.0"/>
    <arg name="uav1_offset_z" default="0.0"/>

    <arg name="uav2_init_x" default="10.0"/>
    <arg name="uav2_init_y" default="-20.0"/>
    <arg name="uav2_init_z" default="3.0"/>
    <!-- <arg name="uav2_init_z" default="20.0"/> -->

	<arg name="uav2_end_x" default="0.0"/>
    <arg name="uav2_end_y" default="0.0"/>
    <arg name="uav2_end_z" default="9.0"/>

    <arg name="uav2_offset_x" default="0.0"/>
    <arg name="uav2_offset_y" default="-3.0"/>
    <arg name="uav2_offset_z" default="0.0"/>

    <arg name="uav3_init_x" default="10.0"/>
    <arg name="uav3_init_y" default="0.0"/>
    <arg name="uav3_init_z" default="9.0"/>
    <!-- <arg name="uav2_init_z" default="20.0"/> -->

	<arg name="uav3_end_x" default="0.0"/>
    <arg name="uav3_end_y" default="-20.0"/>
    <arg name="uav3_end_z" default="3.0"/>

    <arg name="uav3_offset_x" default="3.0"/>
    <arg name="uav3_offset_y" default="0.0"/>
    <arg name="uav3_offset_z" default="0.0"/>


    <!-- 将该参数设置为ROS参数 -->
    <param name="swarm_num_uav" value="$(arg swarm_num_uav)"/>
    
    <node pkg="genco" type="exp_start" name="exp_start" output="screen" launch-prefix="gnome-terminal --tab --">
		<param name="swarm_num_uav" value="$(arg swarm_num_uav)"/>
		<!-- <param name="sim_mode" value="true"/> -->
        <!-- 无人机在全局下的坐标 -->
		<param name="init_pos_1_x" value="$(arg uav1_init_x)"/>
		<param name="init_pos_1_y" value="$(arg uav1_init_y)"/>
		<param name="init_pos_1_z" value="$(arg uav1_init_z)"/>

		<param name="init_pos_2_x" value="$(arg uav2_init_x)"/>
		<param name="init_pos_2_y" value="$(arg uav2_init_y)"/>
		<param name="init_pos_2_z" value="$(arg uav2_init_z)"/>

        <param name="init_pos_3_x" value="$(arg uav3_init_x)"/>
		<param name="init_pos_3_y" value="$(arg uav3_init_y)"/>
		<param name="init_pos_3_z" value="$(arg uav3_init_z)"/>

		

        <!-- 无人机坐标系相对于全局坐标系offset -->
        <param name="offset_pos_1_x" value="$(arg uav1_offset_x)"/>
		<param name="offset_pos_1_y" value="$(arg uav1_offset_y)"/>
		<param name="offset_pos_1_z" value="$(arg uav1_offset_z)"/>

		<param name="offset_pos_2_x" value="$(arg uav2_offset_x)"/>
		<param name="offset_pos_2_y" value="$(arg uav2_offset_y)"/>
		<param name="offset_pos_2_z" value="$(arg uav2_offset_z)"/>

        <param name="offset_pos_3_x" value="$(arg uav3_offset_x)"/>
		<param name="offset_pos_3_y" value="$(arg uav3_offset_y)"/>
		<param name="offset_pos_3_z" value="$(arg uav3_offset_z)"/>

		
	</node>

    <node pkg="genco" type="swarm_offset" name="swarm_offset" output="screen" launch-prefix="gnome-terminal --tab --">
		<param name="swarm_num_uav" value="$(arg swarm_num_uav)"/>
	    
        <!-- 无人机坐标系相对于全局坐标系offset -->
        <param name="offset_pos_1_x" value="$(arg uav1_offset_x)"/>
		<param name="offset_pos_1_y" value="$(arg uav1_offset_y)"/>
		<param name="offset_pos_1_z" value="$(arg uav1_offset_z)"/>

		<param name="offset_pos_2_x" value="$(arg uav2_offset_x)"/>
		<param name="offset_pos_2_y" value="$(arg uav2_offset_y)"/>
		<param name="offset_pos_2_z" value="$(arg uav2_offset_z)"/>

        <param name="offset_pos_3_x" value="$(arg uav3_offset_x)"/>
		<param name="offset_pos_3_y" value="$(arg uav3_offset_y)"/>
		<param name="offset_pos_3_z" value="$(arg uav3_offset_z)"/>

		
    </node>

    

    <!-- 启动rviz,设为false可关闭 -->
	<arg name="visualization" default="true"/>
	<group if="$(arg visualization)">
        <node type="rviz" name="rviz" pkg="rviz" args="-d $(find genco)/exp_view.rviz" />
    </group>

</launch>

